buildSrc restoration in integration

igordmn opened the pull request:
## Release Notes
N/A

MatkovIvan committed:
Replace gradle wrapper to regular public distribution

MatkovIvan committed:
Do not require offline structure from repo-tool checkout

MatkovIvan commented on settings.gradle:
```diff
@@ -958,8 +983,8 @@ includeProject(":navigation:lint:common-test", [BuildType.MAIN, BuildType.FLAN,
 includeProject(":navigation:navigation-runtime-ktx", [BuildType.MAIN, BuildType.FLAN, BuildType.COMPOSE, BuildType.KMP])
-includeProject(":navigation:navigation-safe-args-generator", [BuildType.MAIN, BuildType.FLAN])
-includeProject(":navigation:navigation-safe-args-gradle-plugin", [BuildType.MAIN, BuildType.FLAN])
+//includeProject(":navigation:navigation-safe-args-generator", [BuildType.MAIN, BuildType.FLAN])
```
Modify `shouldIncludeInJetBrainsFork` instead

MatkovIvan commented on settings.gradle:
```diff
@@ -217,11 +220,14 @@ enum BuildType {
 
 private String getRequestedProjectSubsetName() {
+    return "COMPOSE"
+    /* In JetBrains Fork only COMPOSE subset is supported.
```
no-repo changes are not supposed to be merged into the integration branch

MatkovIvan commented on gradle.properties:
```diff
@@ -114,3 +111,38 @@ org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks=**/out/.g
+#   artifactRedirection.version.<redirectionGroup>.<redirectionModule>
+#   covers all sub-module projects
+#   (artifactRedirection.version.androidx.compose covers androidx.compose.*:*)
+artifactRedirection.version.androidx.compose=1.10.0-beta02
```
Can we move it into a separate file to reduce conflicts?

MatkovIvan commented on navigation/navigation-testing/src/jvmTest/kotlin/androidx/navigation/testing/TestSavedStateHandleFactory.kt:
```diff
@@ -244,33 +244,6 @@ class TestSavedStateHandleBuilder {
     }
-
-    @Test
-    fun handleEncodedValues() {
```
Is it related to this PR?

MatkovIvan commented on navigation/navigation-common/src/commonTest/kotlin/androidx/navigation/NavigatorProviderTest.kt:
```diff
@@ -130,13 +130,6 @@ class NavigatorProviderTest {
 internal expect class NoNameNavigator() : Navigator<NavDestination> {
     override fun createDestination(): NavDestination
 
-    override fun navigate(
```
Is it related to this PR?

MatkovIvan commented on lifecycle/lifecycle-viewmodel-navigation3/build.gradle:
```diff
@@ -24,33 +24,42 @@
+    mac()
+    ios()
+    js()
+    wasmJs()
```
Let's merge these changes into integration branch separately

MatkovIvan commented on compose/ui/ui-text/build.gradle:
```diff
@@ -285,3 +276,9 @@ androidx {
+
+afterEvaluate {
+    tasks.withType(Test) { t ->
+        t.jvmArgs += ["--add-opens=java.desktop/sun.font=ALL-UNNAMED"]
```
What's the context around it? Do we have some references/commits with explanation?

MatkovIvan commented on buildSrc/public/src/main/kotlin/org/jetbrains/androidx/build/ComposePlatforms.kt:
```diff
@@ -14,18 +14,14 @@ import org.gradle.api.Project
     WasmJs("Web"),
     MacosX64("Macos"),
     MacosArm64("Macos"),
-    UikitX64("UiKit"), // TODO: Align with AOSP: rename to iOS
```
Let's merge it separately

igordmn commented:
@MatkovIvan, it is not ready for review yet (this code is just to make it work, and will be refactored)

MatkovIvan commented on testutils/testutils-xctest/build.gradle:
```diff
@@ -55,6 +55,12 @@ kotlin {
+        iosMain.languageSettings.languageVersion = KotlinTarget.DEFAULT.apiVersion.version
+        iosX64Main.languageSettings.languageVersion = KotlinTarget.DEFAULT.apiVersion.version
+        iosArm64Main.languageSettings.languageVersion = KotlinTarget.DEFAULT.apiVersion.version
+        iosSimArm64Main.languageSettings.languageVersion = KotlinTarget.DEFAULT.apiVersion.version
```
It's already handled on line 53

MatkovIvan commented on lifecycle/lifecycle-viewmodel-navigation3/build.gradle:
```diff
@@ -48,14 +48,18 @@ androidXMultiplatform {
+                def savedstateVersion = project.findProperty('artifactRedirection.version.androidx.savedstate')
+                api("androidx.savedstate:savedstate:$savedstateVersion")
+                api("androidx.savedstate:savedstate-compose:$savedstateVersion")
+                implementation(libs.kotlinStdlib)
```
Explicit `kotlinStdlib` shouldn't be necessary nowadays

MatkovIvan commented on compose/ui/ui-test/build.gradle:
```diff
@@ -202,6 +202,22 @@ androidXMultiplatform {
+            dependsOn(nativeTest)
+        }
+
+        uikitMain {
```
No more uikitMain in the project

MatkovIvan commented on compose/ui/ui-test/build.gradle:
```diff
@@ -202,6 +202,22 @@ androidXMultiplatform {
             dependsOn(nativeTest)
         }
 
+        macosMain {
```
Duplicate

MatkovIvan commented on compose/runtime/runtime-annotation/gradle.properties:
There is no such file in `jb-main` at this location, please remove it.

Context: this module is not published from fork anymore and we don't have a stub for it

MatkovIvan commented on compose/material3/material3-adaptive-navigation-suite/build.gradle:
```diff
@@ -108,6 +108,7 @@ androidXMultiplatform {
             dependsOn(commonTest)
             dependencies {
                 implementation(project(":kruth:kruth"))
+                implementation(libs.kotlinTest)
```
It's already in dependencies. See `commonTest`

MatkovIvan commented on compose/material3/adaptive/adaptive-layout/build.gradle:
```diff
@@ -120,6 +120,7 @@ androidXMultiplatform {
             dependsOn(commonTest)
             dependencies {
                 implementation(project(":kruth:kruth"))
+                implementation(libs.kotlinTest)
```
It's already in dependencies. See `commonTest`

MatkovIvan commented on buildSrc/public/src/main/kotlin/org/jetbrains/androidx/build/ComposePlatforms.kt:
```diff
@@ -14,8 +14,7 @@ import org.gradle.api.Project
     Desktop("Jvm"),
-    AndroidDebug("Android"),
-    AndroidRelease("Android"),
+    Android("Android"),
```
Why are we removing this Android split? Is it intentional?

MatkovIvan commented on lifecycle/lifecycle-viewmodel-navigation3/build.gradle:
```diff
@@ -68,6 +72,14 @@ androidXMultiplatform {
             }
         }
 
+        jvmMain {
```
Shouldn't `androidMain` depend on this too?

igordmn committed:
no-repo. exclude projects

igordmn committed:
Fix run configuration

igordmn committed:
Remove unused code

igordmn committed:
Introduce sample_font_fork.ttf

Because it sample_font.ttf changed in AOSP, and we would have to adapt test for it

igordmn committed:
API dump

igordmn committed:
Fix buildDir for iOS instrumented tests

igordmn committed:
Fix Version.isSnapshot()

igordmn committed:
Replace AndroidDebug/AndroidRelease by Android

Only Android is supported by new AGP

igordmn committed:
Remove redundant dependencies

igordmn committed:
Fix navigation-compose iOS compilation

@JvmSuppressWildcards isn't in the expect, and not in jb-main

igordmn committed:
Update binaryCompatibilityValidator

igordmn committed:
Restore watchos/tvos publication

igordmn committed:
Fix Web

igordmn committed:
Fix Desktop

igordmn committed:
Fix lifecycle-viewmodel-navigation3

igordmn committed:
Fix publishing tasks

igordmn committed:
Restore fork gradle properties

igordmn committed:
Fix :lifecycle:lifecycle-runtime:compileTestKotlinWasmJs

```
> Task :lifecycle:lifecycle-runtime:compileTestKotlinWasmJs FAILED
e: file:///D:/Work/2/compose-multiplatform-core/lifecycle/lifecycle-runtime/src/wasmJsTest/kotlin/androidx/lifecycle/runLifecycleTest.wasmJs.kt:24:5 Return type mismatch: expected 'JsPromiseInterfaceForTesting', actual 'Promise<JsAny?>'.
e: file:///D:/Work/2/compose-multiplatform-core/lifecycle/lifecycle-runtime/src/webTest/kotlin/androidx/lifecycle/runLifecycleTest.web.kt:23:8 Conflicting overloads:
actual fun runLifecycleTest(block: suspend CoroutineScope.() -> Unit): JsPromiseInterfaceForTesting
```

- runLifecycleTest.wasmJs.kt was added in `integration`
- but isn't compilable
- jb-main doesn't use this file

MatkovIvan committed:
Load `compatibility-stub` projects instead of original ones

MatkovIvan committed:
Update build scripts in stub projects

igordmn committed:
Fix extensions

igordmn committed:
Fix publication

igordmn committed:
Set Java compatibility to 11 for Android and Desktop

igordmn committed:
no-repo. fix android compilation

igordmn committed:
Update build scripts in stub projects

Also Change Android namespaces for stub projects to avoid conflicts

igordmn committed:
Remove runtime-retain project

igordmn committed:
Remove :compose:runtime:runtime-annotation project

igordmn committed:
Restore dependencies to jb-main state

To avoid combining multiple changes

Plan:
- merge to jb-main
- sync dependencies with AOSP in jb-main(TODO create task)

igordmn committed:
Fix Android redirection

igordmn committed:
Fix :compose:ui:ui-uikit error because of parallel execution
